components/list/list-header.js

```diff
@@ -129,28 +163,88 @@ class ListHeader extends RtlMixin(LocalizeCoreElement(LitElement)) {
 	render() {
 		const classes = {
 			'd2l-list-header-container': true,
-			'd2l-list-header-container-slim': (!this._hasActions && !this.selectAllPagesAllowed),
+			'd2l-list-header-container-pre-showing': (this.requiresSelection && this._state === 'pre-showing'),
+			'd2l-list-header-container-showing': (this.requiresSelection && this._state === 'showing'),
+			'd2l-list-header-container-hiding': (this._state === 'hiding')
+		};
+		const classesInner = {
+			'd2l-list-header-container-inner': true,
+			'd2l-list-header-slim': (!this._hasActions && !this.selectAllPagesAllowed),
 			'd2l-list-header-extend-separator': this._extendSeparator
 		};
 		return html`
 			<div class="${classMap(classes)}">
```

Comment:
We need to move the element with `min-height` into an inner element, so that it doesn't mess up height animation. Note: I think the `min-height` was being applied for the case where there are no actions in the header... just select-all. i.e. in order to ensure we have a consistent height regardless of the header contents.